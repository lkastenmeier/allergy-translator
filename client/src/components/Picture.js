import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

const PictureBox = styled.img`
  height: 80%;
  background-color: white;
  padding: 3px;
`;

export default function Picture({ src, alt }) {
  return <PictureBox src={src} alt={alt} />;
}

Picture.propTypes = {
  src: PropTypes.string.isRequired,
  alt: PropTypes.string
};
