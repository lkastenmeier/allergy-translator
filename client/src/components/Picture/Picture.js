//Styling for images in AllergySelection and MyCardBox

//Dependencies
import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

const StyledPicture = styled.img`
  height: 80%;
  background-color: white;
  padding: 3px;
`;

export default function Picture({ src, alt }) {
  return <StyledPicture data-testid="picture" src={src} alt={alt} />;
}

Picture.propTypes = {
  src: PropTypes.string.isRequired,
  alt: PropTypes.string
};
