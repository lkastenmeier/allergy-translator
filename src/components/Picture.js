import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

const PictureBox = styled.img`
  background-color: white;
  height: 111px;
`;

export default function Picture({ src, alt }) {
  return <PictureBox src={src} alt={alt} />;
}
Picture.propTypes = {
  src: PropTypes.string.isRequired,
  alt: PropTypes.string
};
