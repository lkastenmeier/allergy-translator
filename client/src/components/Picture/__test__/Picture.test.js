// Unit test of the Picture Component

//imports
import React from "react";
import ReactDOM from "react-dom";
import Picture from "./../Picture";
import "@testing-library/jest-dom/extend-expect";
import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Picture />, div);
});

it("renders picture correctly", () => {
  const { getByTestId } = render(<Picture alt="alt" src="/icons/MainLogo" />);
  expect(getByTestId("picture").alt).toEqual("alt");
  expect(getByTestId("picture").src.slice(-15)).toEqual("/icons/MainLogo");
});

it("matches snapshot", () => {
  const tree = renderer
    .create(<Picture src="/icons/MainLogo" alt="alt" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
