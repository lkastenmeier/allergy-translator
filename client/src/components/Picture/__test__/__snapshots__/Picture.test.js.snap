// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<img
  alt="alt"
  className="sc-bdVaJa wQRFw"
  data-testid="picture"
  src="/icons/MainLogo"
/>
`;
